dts/bindings/power/stm32,stm32l4-power.yaml

```diff
@@ -0,0 +1,10 @@
+compatible: "stm32,stm32l4-power"
+
+include: zephyr,power-state.yaml
+
+properties:
+  sram-retention:
```

Comment:
That would make sense. I checked it, and multiple MCUs offer SRAM retention in low-power mode. Should I add that property to the `zephyr,power-state` binding?

Reply:
> Should I add that property to the zephyr,power-state binding?

I think so, but I'd like @ceolin to confirm. Or @teburd , @JordanYates maybe ?
Btw, I think `dts/bindings/power` is missing from Power Management section in Maintainers file.

Reply:
@ceolin Ping

Reply:
Wondering if that is not the case of using a different power state.
You could have a different state (or substate) to distinguish whether or not retention is enabled. You can have something like:
```
stop0: state0 {
	compatible = "zephyr,power-state";
	power-state-name = "suspend-to-idle";
	substate-id = <1>;
	min-residency-us = <100>;
};
stop1: state1 {
	compatible = "zephyr,power-state";
	power-state-name = "suspend-to-idle";
	substate-id = <2>; /* This one sram rentetion enabled */
	min-residency-us = <500>;
};
stop2: state2 {
	compatible = "zephyr,power-state";
	power-state-name = "suspend-to-idle";
	substate-id = <3>; /* This one sram rentetion disabled */
	min-residency-us = <500>;
	status = "disabled"; /* SRAM retention is enabled by default */
};

...

void set_mode_stop(uint8_t substate_id)
{
	...
	case 2:
		LL_PWR_EnableSRAM3Retention();
		...
		break;
	case 3:
		LL_PWR_DisableSRAM3Retention();
		...
		break;
	..
}
```


Basically you can have two states with the same parameters but different substates but only of them enabled to not confuse the policy.

Reply:
Ok, thanks for the feedback. 
That's indeed a good point and would allow to support them simultaneously.
Drawback is the increased footprint
@coran21 What do you think ?

Reply:
I'm not sure if I understand it correctly. 

Would be ok to add the new state like this?

```			
stop2_sram_en: state2_sram_en {
	compatible = "zephyr,power-state";
	power-state-name = "suspend-to-idle";
	substate-id = <4>; /* new substate with SRAM retention on */
	min-residency-us = <1000>;
	status = "okay"; /* SRAM retention is enabled by default */
};			

cpu0: cpu@0 {
	cpu-power-states = <&stop0 &stop1 &stop2 &stop2_sram_en>; 
};
```
I understand the changes in `void set_mode_stop(uint8_t substate_id)`  function.

Reply:
> Would be ok to add the new state like this?

That's indeed what is proposed by @ceolin.

Though, wait a bit, I'm having second thoughts on this:
The intent here is to configure SRAM3 retention, which is not the main ram. I'm worried that adding a dedicated state for "RAM retention" would confuse users as we're only playing with a secondary RAM here.
One idea would be to work on the state naming: `stop2_sram3_en`, but since we're talking about a secondary RAM, we can actually have multiple of those sram2, ... sramN. Adding a specific state for each wouldn't scale and even if, what about various combinations..

Thinking down that road further, I think the right thing to do would be to add a dt property to the concerned sram node itself.

@ceolin Does it make sense to you ?